api/lib/models/checkout-items/checkout-items-data-mapper-impl.ts

```diff
@@ -68,13 +60,90 @@ export class CheckoutItemsDataMapperImpl extends GenericDataMapper
     ).toPromise();
   }
 
-  public getAllAvailable(): Promise<IDbResult<Stream<CheckoutItems>>> {
+  public get(id: UidType, opts?: IUowOpts): Promise<IDbResult<CheckoutItems>> {
+    let queryBuilder = squel.select({ 
+      autoQuoteFieldNames: true, 
+      autoQuoteTableNames: true 
+    })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder
+      .where(`${this.pkColumnName}= ?`, id);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<CheckoutItems>(query.text, query.values, { stream: false, cache: true }))
+      .pipe(
+        map((checkoutItems: CheckoutItems) => ({ result: 'Success', data: checkoutItems })),
+      )
+      .toPromise();
+  }
+
+  public async getAll(opts?: IUowOpts): Promise<IDbResult<Stream<CheckoutItems>>> {
+    let queryBuilder = squel.select({
+      autoQuoteFieldNames: true,
+      autoQuoteTableNames: true,
+    })
+      .from(this.tableName, 'checkoutItems');
+      if(opts && opts.fields) {
+        queryBuilder = queryBuilder.fields(opts.fields);
+      }
+      if(opts && opts.startAt) {
+        queryBuilder = queryBuilder.offset(opts.startAt);
+      }
+      if(opts && opts.count) {
+        queryBuilder = queryBuilder.limit(opts.count);
+      }
+      if (opts && opts.byHackathon) {
```

Comment:
Are checkout items supported hackathon wise? I thought only the CheckoutObject class had a hackathon reference.

Reply:
Going off the old js model, the "available" methods depend on the hackathon but I messed up the query and made the normal "get"s depend on the hackathon instead of the available ones. If you want this can be changed to not support hackathons.

Reply:
If the join by hackathon is performed on this the query will fail. It should be removed.